examples/Azure_IoT_Central_ESP32/Azure_IoT_Central_ESP32.ino

```diff
@@ -406,14 +411,22 @@ void loop()
 {
   if (WiFi.status() != WL_CONNECTED)
   {
-    azure_iot_stop(&azure_iot);
+    if (azure_iot.state != azure_iot_state_not_initialized)
+      azure_iot_stop(&azure_iot);
```

Comment:
Within `azure_iot_stop` the first check is if `azure_iot->state == azure_iot_state_not_initialized`.
If `azure_iot` is `azure_iot_state_not_initialized`, nothing is done by `azure_iot_stop`.
So the only reason to check here again is to avoid printing error logs? Remember that this is a sample, so for a production scenario customers can just remove the logging from azure_iot_stop.